### PR TITLE
[Merged by Bors] - chore: Delete `Data.Nat.Units`, `Data.Int.Units`

### PR DESCRIPTION
**Pull request by YaelDillies**
The lemmas can easily move to `Algebra.Group.Nat` and `Algebra.Group.Int`, `Algebra.Ring.Int` respectively.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/GCDMonoid/Nat.lean [hunk 1]
```diff
@@ -5,7 +5,7 @@ Authors: Johannes Hölzl, Jens Wagemaker, Aaron Anderson
 -/
 import Mathlib.Algebra.GCDMonoid.Basic
 import Mathlib.Algebra.Ring.Nat
-import Mathlib.Data.Nat.Units
+import Mathlib.Algebra.Group.Nat
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/GCDMonoid/Nat.lean:8]: Order alphabetically, please

File: Mathlib/Algebra/Group/Int.lean [hunk 2]
```diff
@@ -95,6 +96,106 @@ lemma toAdd_zpow (a : Multiplicative ℤ) (b : ℤ) : toAdd (a ^ b) = toAdd a *
...
+
+-- theorem units_eq_one_or (u : ℤˣ) : u = 1 ∨ u = -1 := by
+--   simpa only [Units.ext_iff, units_natAbs] using natAbs_eq u
+-- #align int.units_eq_one_or Int.units_eq_one_or
+
+-- theorem units_ne_iff_eq_neg {u u' : ℤˣ} : u ≠ u' ↔ u = -u' := by
+--   rcases units_eq_one_or u with rfl | rfl <;>
+--   rcases units_eq_one_or u' with rfl | rfl <;>
+--   decide
+
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Group/Int.lean:124]:
```suggestion
```
if these really can't go here, but I'd like to understand why not

File: Mathlib/AlgebraicTopology/SimplicialSet.lean [hunk 3]
```diff
@@ -7,7 +7,7 @@ import Mathlib.AlgebraicTopology.SimplicialObject
 import Mathlib.CategoryTheory.Limits.Shapes.Types
 import Mathlib.CategoryTheory.Yoneda
 import Mathlib.Data.Fin.VecNotation
-import Mathlib.Data.Nat.Units
+import Mathlib.Algebra.Group.Nat
```
Comment (Ruben-VandeVelde) [inline Mathlib/AlgebraicTopology/SimplicialSet.lean:10]: Order

File: Mathlib/Data/Fintype/Units.lean [hunk 4]
```diff
@@ -5,7 +5,7 @@ Authors: Mario Carneiro
 -/
 import Mathlib.Data.Fintype.Prod
 import Mathlib.Data.Fintype.Sum
-import Mathlib.Data.Int.Units
+import Mathlib.Algebra.Ring.Int
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Fintype/Units.lean:8]: Order

File: Mathlib/Data/Int/AbsoluteValue.lean [hunk 5]
```diff
@@ -6,7 +6,7 @@ Authors: Anne Baanen
 import Mathlib.Algebra.Module.Defs
 import Mathlib.Algebra.Order.AbsoluteValue
 import Mathlib.Data.Int.Cast.Lemmas
-import Mathlib.Data.Int.Units
+import Mathlib.Algebra.Ring.Int
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Int/AbsoluteValue.lean:9]: You guessed it: order

File: Mathlib/Data/Ordmap/Ordset.lean [hunk 6]
```diff
@@ -6,7 +6,7 @@ Authors: Mario Carneiro
 import Mathlib.Algebra.Order.Ring.Defs
 import Mathlib.Algebra.Group.Int
 import Mathlib.Data.Nat.Dist
-import Mathlib.Data.Nat.Units
+import Mathlib.Algebra.Group.Nat
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Ordmap/Ordset.lean:9]: Here too

**Review by Ruben-VandeVelde — COMMENTED** [review]
lgtm modulo the location of units_eq_one_or/units_ne_iff_eq_neg

File: Mathlib/Algebra/Group/Int.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Group/Int.lean:124]: The `Neg ℤˣ` instance requires the `Ring ℤ` instance. In contrast, talking about elements of `ℤ` respecting `IsUnit` only requires the `CommMonoid ℤ` and `Neg ℤ` instances.

**Review by Ruben-VandeVelde — APPROVED** [review]
I see.

maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**